src/cli/Hibp.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Copyright (C) 2019 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdlib>
+#include <cstring>
+#include <stdio.h>
+
+#include "Hibp.h"
+#include "cli/Utils.h"
+
+#include <QCommandLineParser>
+#include <QFile>
+
+#include "cli/TextStream.h"
+#include "core/Database.h"
+#include "core/Entry.h"
+#include "core/Group.h"
+#include "core/Tools.h"
+#include "crypto/CryptoHash.h"
+
+namespace
+{
+
+    struct Sha1Hash
+    {
+        uchar m_data[20];
```

Comment:
Please do not use char arrays. Use QByteArrays. I don't see why this struct is needed at all. A simple QByteArray should be enough. You are even constructing this object from a QByteArray.

Reply:
As mentioned [in my comment above](#discussion_r257463912), I did it this way for performance reasons, as I wanted to avoid dynamic memory allocations.

However, I haven't actually benchmarked that so far. I'll do that next, then we can decide based on the results.

Reply:
You do not need to reallocate anything if you reuse the same QByteArray.